src/main/java/io/jenkins/plugins/credentials/secretsmanager/config/Client.java

```diff
@@ -63,9 +79,41 @@ public void setRegion(String region) {
         this.region = Util.fixEmptyAndTrim(region);
     }
 
+    static Optional<ProxyConfiguration> getProxyConfiguration() {
+        final var maybeProxyConfiguration = Jenkins.lookup(ProxyConfiguration.class);
```

Comment:
Ah, I was trying to avoid the use of `Jenkins.get()` if I could because this can cause runtime errors when run in certain conditions. Unfortunately the alternative does not seem to have worked, so we will have to use `Jenkins.get()`. Will push an update

Reply:
Ok, once it's updated I can test it again. Thanks

Reply:
Update pushed - ready to test

Reply:
From my testing of the latest change I can see the plugin works as expected. Thanks

Reply:
Great - just a couple of other things:

1. Have you encountered any kind of TLS certificate errors when proxying traffic from the plugin? (The AWS SDK might possibly do certificate pinning, or have other measures to resist proxy servers decrypting traffic to AWS.)
2. Could you try applying your proxy settings just in the plugin's Client Configuration (i.e. not in the system-wide Jenkins proxy setting) to see what happens?
3. Just to be very sure that this change does not break the default non-proxy connection flow, could you try the plugin in a non-proxy version of Jenkins to ensure that works?

Reply:
1. No, I have not seen any issues.
2. Yes, I have tried both.
  Global:
    
![image](https://github.com/jenkinsci/aws-secrets-manager-credentials-provider-plugin/assets/2352273/8ed0e782-c931-474b-bf88-31d7365eb8ff)

![image](https://github.com/jenkinsci/aws-secrets-manager-credentials-provider-plugin/assets/2352273/b30a66f3-58ad-40ec-9a44-1c8827f32606)

![image](https://github.com/jenkinsci/aws-secrets-manager-credentials-provider-plugin/assets/2352273/535364f6-38e1-4676-96e5-2a302015195c)

Client Configuration:

![image](https://github.com/jenkinsci/aws-secrets-manager-credentials-provider-plugin/assets/2352273/bf30ec69-70dd-46b8-a383-1221a7f0a854)

![image](https://github.com/jenkinsci/aws-secrets-manager-credentials-provider-plugin/assets/2352273/c9be99f1-2638-4b8f-a437-42a34bdc9909)

![image](https://github.com/jenkinsci/aws-secrets-manager-credentials-provider-plugin/assets/2352273/6a5f25c4-644b-4cd3-8fbb-fe3c77db1a9b)

![image](https://github.com/jenkinsci/aws-secrets-manager-credentials-provider-plugin/assets/2352273/4864ca1e-7ebd-4a71-b50a-4f8cc443ab03)

3. Yes, I have tested that using moto and localhost so Jenkins doesn't have to go through the proxy

![image](https://github.com/jenkinsci/aws-secrets-manager-credentials-provider-plugin/assets/2352273/f5d231f8-5ac0-4ad6-afaf-1bc282827848)

![image](https://github.com/jenkinsci/aws-secrets-manager-credentials-provider-plugin/assets/2352273/5aa16075-30cf-4dfc-9a84-593b1d81d72c)

![image](https://github.com/jenkinsci/aws-secrets-manager-credentials-provider-plugin/assets/2352273/2fb0d0e4-7d2f-4b96-8220-6e51e7423ad9)

Reply:
Great 👍

Reply:
Thanks for your help. What are the release plans for the next version of the plugin?

Reply:
The next version has been released already, so if you go to the Jenkins Update Center you should see it available to use.